components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2243,22 +2252,39 @@ private AssociatedRolesConfig getAssociatedRoles(String applicationId, Connectio
-            try (ResultSet resultSet = preparedStatement.executeQuery()) {
-                while (resultSet.next()) {
-                    associatedRoleIds.add(resultSet.getString(1));
+        String allowedAudience =
+                getSPPropertyValueByPropertyKey(applicationId, ALLOWED_ROLE_AUDIENCE_PROPERTY_NAME, tenantDomain);
+        if (RoleConstants.APPLICATION.equalsIgnoreCase(allowedAudience)) {
+            try (NamedPreparedStatement preparedStatement = new NamedPreparedStatement(connection,
+                    ApplicationMgtDBQueries.LOAD_ASSOCIATED_ROLES)) {
+                preparedStatement.setString(ApplicationMgtDBQueries.SQLPlaceholders.DB_SCHEMA_COLUMN_NAME_APP_ID,
+                        applicationId);
+                try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                    while (resultSet.next()) {
+                        associatedRoleIds.add(resultSet.getString(1));
+                    }
                 }
+
+                associatedRolesConfig.setRoles(buildAssociatedRolesWithRoleName(associatedRoleIds, tenantDomain));
+            } catch (SQLException | IdentityRoleManagementException e) {
+                throw new IdentityApplicationManagementException(
+                        "Error while retrieving associated roles for application ID: " + applicationId, e);
             }
-            associatedRolesConfig.setRoles(buildAssociatedRolesWithRoleName(associatedRoleIds, tenantDomain));
-        } catch (SQLException | IdentityRoleManagementException e) {
-            throw new IdentityApplicationManagementException(
-                    "Error while retrieving associated roles for application ID: " + applicationId, e);
+        } else if (RoleConstants.ORGANIZATION.equalsIgnoreCase(allowedAudience)) {
+            RoleManagementService roleManagementServiceV2 = ApplicationManagementServiceComponentHolder.getInstance()
+                    .getRoleManagementServiceV2();
+            try {
+                List<RoleBasicInfo> listOfAllRoles = roleManagementServiceV2.getRoles(AUDIENCE_EQ_ORGANIZATION,
+                        0, 0, null, null, tenantDomain);
+                List<String> roleIds = listOfAllRoles.stream().map(RoleBasicInfo::getId).collect(Collectors.toList());
+                associatedRolesConfig.setRoles(buildAssociatedRolesWithRoleName(roleIds, tenantDomain));
+            } catch (IdentityRoleManagementException e) {
+                throw new IdentityApplicationManagementException("Error while retrieving associated roles for " +
+                        "application ID: " + applicationId, e);
+            }
+
         }
-        String allowedAudience =
-                getSPPropertyValueByPropertyKey(applicationId, ALLOWED_ROLE_AUDIENCE_PROPERTY_NAME, tenantDomain);
+
         associatedRolesConfig.setAllowedAudience(
                 StringUtils.isNotBlank(allowedAudience) ? allowedAudience.toLowerCase() : RoleConstants.ORGANIZATION);
```

Comment:
we set the default audience as organization for the application with these changes?